lib/imagec/docker.go

```diff
@@ -146,18 +149,22 @@ func LearnAuthURL(options ImageCOptions) (*url.URL, error) {
 }
 
 // FetchToken fetches the OAuth token from OAuth endpoint
-func FetchToken(url *url.URL) (*Token, error) {
+func FetchToken(ic *ImageC, url *url.URL, progressOutput progress.Output) (*Token, error) {
```

Comment:
It looks like the Learn* functions are already using a context timeout by way of `fetcher.Head`.

FetchImageBlob is getting the context that the `Transfer` provides, so I wrap that in a context using our default timeout. For everything else, I've created a context using our default timeout in `PullImage` and I send that in to all the `Fetch*` functions from there. Sound ok?